packages/graphql/src/translate/queryAST/utils/is-nested-context.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type Cypher from "@neo4j/cypher-builder";
+import type { QueryASTContext } from "../ast/QueryASTContext";
+
+/** Checks if provided context has the target field defined (which would make it a nested translation context) */
+export function isNestedContext(context: QueryASTContext): context is QueryASTContext & { target: Cypher.Node } {
+    return context.target !== undefined;
+}
```

Comment:
Are we sure this is correct?

At the moment, nestedContext happens when there is a source. A target is always defined (except on the new cases). Has this changed that much?

Reply:
Sorry didn't have time to fully review yet, but I  wanted to raise this as well, if you look at the `propertyFilter` file changes, you'll find for instance:

```typescript
private getPropertyRef(queryASTContext: QueryASTContext): Cypher.Property {
    if (this.attachedTo === "node") {
        if (!isNestedContext(queryASTContext)) throw new Error("No parent node found!");
        return queryASTContext.target.property(this.attribute.databaseName);
    } else if (this.attachedTo === "relationship" && queryASTContext.relationship) {
        return queryASTContext.relationship.property(this.attribute.databaseName);
    } else {
        throw new Error("Transpilation error");
    }
}
```

The assertion line:
```typescript
if (!isNestedContext(queryASTContext)) throw new Error("No parent node found!");
```
Give the sense that a PropertyFilter can be translated only with "nested contexts" although this is not true, for normal operations like Read/Connection, in fact, the context passed to a top-level Read Operation is directly passed to the filters without creating a "nested context", even if the `isNestedContext` returns true because the target, in that case, is defined. I feel it's worth changing this name to a less opinionated name without inferring that if the target is defined then the context is "nested"!

Reply:
mmm yes sure, good points! I was mistaken thinking that target being undefined means you are in a nestedContext, it's more of the other way around sort of.
Would `hasTarget` work or do you have any suggestions?

Reply:
Works for me!